handlers/clusters.go

```diff
@@ -442,6 +440,26 @@ func CreateClusterResponse(params clusters.CreateClusterParams) middleware.Respo
 	// TODO ties into cluster status, make a note if the service is missing
 	sc.Create(&websv.Service)
 
+	// Wait for the replication controllers to exist before building the response.
+	rcc := client.ReplicationControllers(namespace)
+	{
+		var mrepl, wrepl *kapi.ReplicationController
+		mrepl = nil
+		wrepl = nil
+		for i := 0; i < 4; i++ {
+			if mrepl == nil {
+				mrepl, _ = getReplController(rcc, clustername, masterType)
+			}
+			if wrepl == nil {
+				wrepl, _ = getReplController(rcc, clustername, workerType)
+			}
+			if wrepl != nil && mrepl != nil {
+				break
+			}
+			time.Sleep(250 * time.Millisecond)
+		}
```

Comment:
I don't think we need one, those are both local to the block.  The attempt here was to delay until the rcs were created, without going over a second.  If they're still nil at the end of the block we assume that kube will do the right thing since there was no error on the create of the dcs earlier, and the situation is no worse than it was without the (slight) fix.